benchmark/profiliable_nonhydrostatic_simulation.jl

```diff
@@ -0,0 +1,45 @@
+push!(LOAD_PATH, joinpath(@__DIR__, ".."))
+
+using Profile
+using CUDA
+using Oceananigans
+using Benchmarks
+
+# Benchmark parameters
+
+timestepper = :QuasiAdamsBashforth2
+Arch = CPU
+FT = Float64
+N = 128
+
+print_system_info()
+
+# Define benchmarks
+
+@info "Setting up benchmark: ($Arch, $FT, $N)..."
+
+grid = RegularRectilinearGrid(FT, size=(N, N, N), extent=(1, 1, 1))
+
+model = NonhydrostaticModel(architecture = Arch(),
+                            timestepper = timestepper,
+                            grid = grid,
+                            advection = WENO5(),
+                            tracers = :b,
+                            buoyancy = BuoyancyTracer(),
+                            closure = IsotropicDiffusivity(ν=1, κ=1))
+
+simulation = Simulation(model, Δt=1, stop_iteration=1)
+
+@info "warming up"
+
+run!(simulation)
+
+simulation.stop_iteration += 10
+
+if model.architecture isa GPU
+    CUDA.@profile run!(simulation)
+else
+    @profile run!(simulation)
```

Comment:
Outputs and formats profiling results into a sorted flat (not tree) text table. Saves into text file. Could also put in `mincount=100` to shorten outputs by removing negligible results.
```suggestion
    Profile.clear()
    @profile run!(simulation)
    open("nonhydrostatic_profile.txt", "w") do io
        Profile.print(IOContext(io, :displaysize => (1000, 350)), format=:flat, sortedby=:count)
    end
```

Reply:
What about using `StatProfilerHTML`? I got some mileage out of this on #1923 .

Reply:
https://github.com/tkluck/StatProfilerHTML.jl

Reply:
Neat! That's definitely a very nice way to output interactive flame graphs. I'll give it a try for this script and something for  #1919 and #1923. We might be able to embed or link the whole output HTML file directly onto github comments.